components/src/main/java/org/wso2/carbon/messaging/CloseCarbonMessage.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.messaging;
+
+/**
+ * {@link CarbonMessage} type for Close Messages.
+ */
+public class CloseCarbonMessage extends CarbonMessage {
```

Comment:
The problem with CloseCarbonMessage is, it is not going inline with rest of the Message Types (Binary, Text, etc.). 
Renaming it to a more generic one or enclosing it with a more generic one (like what you did for Ping and Pong i.e. Ping + Pong --> ControlMessage) is what I am suggesting.